### PR TITLE
Create updated Ubuntu ECP image/CI job

### PR DESCRIPTION
**Pull request by willdunklin**
Adds a new ECP test image based on the ecp ubuntu20.04 runner

### CODE REVIEW

**Comment by willdunklin** [comment]
Just wanted to note that the CI approval is hanging since the `fedora35-ecp-catalyst` job was removed. I can add the job back into the PR but the Ubuntu image is meant to supplant the older fedora image.

**Comment by burlen** [comment]
20.04 is pretty out dated, since we are making the effort to update this why use ubuntu 22.04, already a year old. I've had to move to this in my development systems to get the latest ROCm release (needed for asynchrounous data transfer). Similarly I had to go to Fedora 37 to get the latest CUDA 12 release because of compiler bugs in 11.

**Comment by willdunklin** [comment]
@burlen
The image is based on the E4S runner image, which is related to the environments the ecp sdk will be ran in. The target HPC systems will be stuck on 20.04 for quite some time, so for the sake of this integration test, it might be best to stick with this version.

**Comment by burlen** [comment]
Feel free to merge this once Ryan is set

**Comment by willdunklin** [comment]
I've wrapped up the changes for this PR (@kwryankrattiger previously reviewed this with me, although another look would be appreciated).

One note is that I needed to use ADIOS2 2.8.3 for the tests to pass, see #133 for info about the failing tests. The Ubuntu image should have parity with the old Fedora one now.

File: .github/ci/docker/ubuntu/install-deps.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    spack mirror add sensei /sensei/buildcache && \
+    spack buildcache update-index sensei
+} || {
+    echo "Error: Failed to install buildcache."
+    echo "  Skipping buildcache installation..."
+}
+
+# install
+N_THREADS=$(grep -c '^processor' /proc/cpuinfo)
+spack install -v --use-cache --no-check-signature -j ${N_THREADS}
```
Comment (kwryankrattiger) [inline .github/ci/docker/ubuntu/install-deps.sh:18]: I think for this we also want to add `--only dependencies`

Comment (willdunklin) [reply .github/ci/docker/ubuntu/install-deps.sh:18]: Ah, actually I think we want to remove the next line, considering this should also install an up to date lmod. Is the splitting between dependencies and the packages important for this script?

**Review by kwryankrattiger — APPROVED** [review]
I think this is okay.

**Pull request merged**